src/codeu/chat/client/commandline/Chat.java

```diff
@@ -40,6 +43,44 @@ public Chat(Context context) {
 		this.panels.push(createRootPanel(context));
 	}
 
+	// RECREATE 
+	//
+	// Recreates the server's last known state to display for the client 
+	// upon relaunching the client 
+	// Will not display anything if there was no information saved 
+	//
+	//
+	public void recreate(Context context) {
+		Hashtable<Uuid, String> allUsers = new Hashtable<Uuid, String>();
+
+		if (context.allUsers().iterator().hasNext()) {
+
+			Iterator<UserContext> users = context.allUsers().iterator(); 
+			UserContext user;
+			System.out.format("USERS: \n");
+			while (users.hasNext()){
+				user = users.next(); 
+				System.out.format("\t\"%s\" Added at %s || UUID: %s\n", 
+				user.user.name, user.user.creation.HMtime(), user.user.id);
+				allUsers.put(user.user.id, user.user.name); 
+				if (!users.hasNext()){
```

Comment:
right now it seems like all the conversations are saved under every user, but you're right--I changed the code so now every conversation is saved into a hashmap and the function prints from there to make sure every conversation is covered